### ISSUE TITLE
Neomutt Can't write to maildir folders anymore

### ISSUE THREAD

**Issue by bluecub** [neomutt/neomutt#1953]
## Expected Behaviour

Usually after deleting and pressing '$' it will usually sync with disk. Now it just doesn't remove deleted emails.

## Actual Behaviour

It gives this error message on the status bar
Unable to write /home/laptop/.config/mutt/email/Inbox

## Steps to Reproduce

Since I upgraded to the latest version

## How often does this happen?

- Always

## When did it start to happen?

- When I upgraded
  Which version did you use to use? I've been using neomutt for more than 2 years and this is the first time I see this behaviour.

## NeoMutt Version

```
NeoMutt 20191102
Copyright (C) 1996-2016 Michael R. Elkins and others.
NeoMutt comes with ABSOLUTELY NO WARRANTY; for details type 'neomutt -vv'.
NeoMutt is free software, and you are welcome to redistribute it
under certain conditions; type 'neomutt -vv' for details.

System: FreeBSD 12.1-RELEASE (amd64)
ncurses: ncurses 5.9.20140222 (compiled with 5.9.20140222)
libiconv: 1.14
GPGme: 1.13.1
hcache backends: tokyocabinet, lmdb

Compiler:
FreeBSD clang version 6.0.1 (tags/RELEASE_601/final 335540) (based on LLVM 6.0.1)
Target: x86_64-unknown-freebsd12.0
Thread model: posix
InstalledDir: /usr/bin

Configure options: --with-ssl=/usr --locales-fix --mandir=/usr/local/man --enable-gpgme --gss --disable-idn --lmdb --enable-nls --enable-notmuch --sasl --with-ui=ncurses --tokyocabinet

Compilation CFLAGS: -O2 -pipe  -fstack-protector-strong -isystem /usr/local/include -fno-strict-aliasing  -std=c99 -D_ALL_SOURCE=1 -D_GNU_SOURCE=1 -D__EXTENSIONS__ -I/usr/local/include -DNCURSES_WIDECHAR -I/usr/include -I/usr/include 

Default options:
  +attach_headers_color +compose_to_sender +compress +cond_date +debug 
  +encrypt_to_self +forgotten_attachments +forwref +ifdef +imap +index_color 
  +initials +limit_current_thread +multiple_fcc +nested_if +new_mail +nntp +pop 
  +progress +quasi_delete +regcomp +reply_with_xorig +sensible_browser +sidebar 
  +skip_quoted +smtp +status_color +timeout +tls_sni +trash 

Compile options:
  -autocrypt +bkgdset +color +curs_set +fcntl -flock -fmemopen +futimens 
  +getaddrinfo -gnutls +gpgme +gss +hcache -homespool -idn -inotify 
  +locales_hack -lua +meta -mixmaster +nls +notmuch +openssl +pgp +sasl +smime 
  -sqlite +start_color +sun_attachment +typeahead 
MAILPATH="/var/mail"
PKGDATADIR="/usr/local/share/neomutt"
SENDMAIL="/usr/sbin/sendmail"
SYSCONFDIR="/usr/local/etc"

To learn more about NeoMutt, visit: https://neomutt.org
If you find a bug in NeoMutt, please raise an issue at:
    https://github.com/neomutt/neomutt/issues
or send an email to: <neomutt-devel@neomutt.org>
```

## Extra Info

* Operating System and its version
FreeBSD
* Were you using multiple copies of NeoMutt at once?
No
* Were you using 'screen' or 'tmux'?
No
* Is your email local (maildir) or remote (IMAP)?
maildir

**Comment by flatcap** [comment]
Sorry about that, @bluecub 
I fixed a bug a couple of days ago that might have been causing this.

If you're technical and are happy to build your own NeoMutt, please can you give the latest version a try (`master`).
If not, don't worry.  There's a new bug-fix release due in a couple of days.

**Comment by bluecub** [comment]
Thanks so much @flatcap 
I will try do that, is there any thing special I need to check to get it working for freeBSD?

**Comment by flatcap** [comment]
[According to the maintainer](https://neomutt.org/distro/freebsd), nothing special.

**Comment by bluecub** [comment]
Perfect I will work on it tomorrow, it a bit late here now :) Thanks again

**Comment by bluecub** [comment]
Hi @flatcap 
I built neomutt from master branch, and now I don't get that specific error. Yet I feel it is not yet there, as I noticed some weird behaviour and here is what I did:

- I deleted email and synced "$" (purged deleted emails). 

- I checked the webmail I can tell they have been deleted, 

- if I change folder and come back to inbox the purged emails will go, 

- yet If I stay in the same folder (inbox for example where I deleted and purged the emails) they don't disappear as it used to happen previously.

Not sure if I was clear enough in describing the issue.
Thanks

**Comment by bluecub** [comment]
I can still see the previous error still there... it is not working properly. Thanks

**Comment by gahr** [comment]
Hi @bluecub , I have been trying to replicate the (mis)behaviour you reported, but to no avail. Can you please try with an empty config file? `neomutt -F /dev/null -f /path/to/your/maildir`

**Comment by bluecub** [comment]
Hi @gahr 
I switched back to the old version, feel free to close this thread if you can't reproduce the error. I will try again once the new version has been released and work with you guys in case I had the same issues. Thanks so much

**Comment by bluecub** [comment]
> Hi @bluecub , I have been trying to replicate the (mis)behaviour you reported, but to no avail. Can you please try with an empty config file? `neomutt -F /dev/null -f /path/to/your/maildir`

Hi @gahr @flatcap 
I tried your trick and worked, I would assume something in my configuration file is not valid anymore?
What I can do to drill down to the root cause and fix the configuration?

Thanks

**Comment by gahr** [comment]
Good to hear that an empty config works. I guess you'll have to add back your config chunk by chunk until it breaks. I don't have a fancier way to do that.

**Comment by mhinsch** [comment]
I have just encountered the same problem it seems (master branch from two days ago). 

Context:
I switched between different maildir folders, sorting by score, removing flags and saving messages to different folders.

Symptoms:
* any write information to the currently selected maildir fails (including quitting neomutt), debug shows return value for mbox_sync == -1
* it is not possible any more to sort the current maildir (nothing happens, no debug message)

I will try to find a way to reliably reproduce the bug.
